src/main/cpp/drivers/adapter.cpp

```diff
@@ -66,8 +66,10 @@ namespace DriverAdapter {
 
 #if DRIVER == 0
         // OpenSL
-        android_CloseAudioDevice( driver_openSL );
-        delete driver_openSL;
+        if ( driver_openSL != nullptr ) {
+            android_CloseAudioDevice(driver_openSL);
```

Comment:
Do not close device if it wasn't allocated. _android_CloseAudioDevice_ is written in C and uses its internal check by comparing against _NULL_ (integer type). This might cause a mismatch between the C++ type _nullptr_ (which is the same value, but treated of the pointer type). Not taking chances here.